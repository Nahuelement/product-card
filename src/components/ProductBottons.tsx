import {  useContext} from 'react';
import { ProductContext } from './ProductCard';
import styles from '../styles/styles.module.css'
import React from 'react'
// import { useDispatch, useSelector } from 'react-redux'

export interface PropsButton {
    className?:string,
    style?:React.CSSProperties,
    product?:any,
    quantity?:number
}

export const ProductBottons = ({className,style,product,quantity}:PropsButton) =>{

    
    const {counter, increaseBy,isMaxCountReached } = useContext(ProductContext)
   console.log(product,quantity)

    // const dispatch = useDispatch()
    

    // const isMaxCount = useCallback(() => {
    //    if(counter===maxCount) return true
    //    else return false
    //  }, [counter,maxCount]);
     
    
    
   return(
       <div 
       className = {`${styles.buttonsContainer} ${className}`}
       style = {style}

       >
       <button 
       className ={  styles.buttonMinus}
     
       
       onClick= {() =>{
        // dispatch({type:'put',payload:product})
        increaseBy(-1)}
       } 
       >
           -
       </button>
       <div 
       className ={styles.countLabel}>
           {counter}
       </div>
       
       <button 
       className ={ ` ${ styles.buttonAdd } ${isMaxCountReached && styles.disabled}  ` }
        disabled = {isMaxCountReached}
       onClick= {() => {
        //    dispatch({type:'add',payload:product})
           increaseBy(+1)}}
       >
           +
       </button>

   </div>
   )
}