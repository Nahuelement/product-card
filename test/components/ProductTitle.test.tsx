import React from 'react'
import  renderer  from 'react-test-renderer'
import { ProductTitle } from '../../src/components';
import { product1 } from '../data/product';
import { ProductCard } from '../../src/components/ProductCard';



describe('ProductTitle', () => {


     test('Mostrar correctamente el titulo', () => { 
         const wrapper = renderer.create(
             <ProductTitle title='Custom Title' />
         )
         expect(wrapper.toJSON() ).toMatchSnapshot();
      
     }) 
     test('debe de mostrar el componente con el nombre del producto', () => { 

        const wrapper = renderer.create(
            <ProductCard product={product1} >
                {
                    ()=>(
                    <ProductTitle title='Custom Title'/> 
                    )
                }
            </ProductCard>
        )
        expect(wrapper.toJSON() ).toMatchSnapshot();

      })
    })