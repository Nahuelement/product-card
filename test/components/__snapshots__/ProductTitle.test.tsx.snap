// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Mostrar correctamente el titulo 1`] = `
<span
  className="productDescription undefined"
>
  Custom Title
</span>
`;

exports[`ProductTitle debe de mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined "
>
  <span
    className="productDescription undefined"
  >
    Custom Title
  </span>
</div>
`;
